File: P1704 Determine if String Halves Are Alike/Solution.java
class Solution {
    // Time: O(n), where n is the length of s
    // Space: O(1)
    public boolean halvesAreAlike(String s) {
        int firstHalfCount = 0;
        int secondHalfCount = 0;
        int strLength = s.length();
        String vowels = "aeiouAEIOU";
        for(int i = 0; i < strLength; i++) {
            if(vowels.indexOf(s.charAt(i)) != -1) {
                if(i < strLength/2) {
                    firstHalfCount++;
                } else {
                    secondHalfCount++;
                }
            }
        }
        return firstHalfCount == secondHalfCount;
    }
}

File: P1704 Determine if String Halves Are Alike/solution.py
class Solution:
    # Time: O(n), where n is the length of s
    # Space: O(1)
    def halvesAreAlike(self, s: str) -> bool:
        first_half_count = 0
        second_half_count = 0
        vowels = "aeiouAEIOU"
        for i in range(len(s)):
            if s[i] in vowels:
                if i < len(s) / 2:
                    first_half_count += 1
                else:
                    second_half_count += 1
        return first_half_count == second_half_count
